src/indexed_crate.rs

```diff
@@ -148,3 +156,73 @@ fn collect_public_items<'a>(
         Visibility::Crate | Visibility::Restricted { .. } => {}
     }
 }
+
+fn create_dummy_trait_items(crate_: &Crate) -> HashMap<Id, Item> {
+    let paths = crate_
+        .index
+        .values()
+        .map(|item| &item.inner)
+        .filter_map(|item_enum| match item_enum {
+            rustdoc_types::ItemEnum::Impl(impl_) => Some(impl_),
+            _ => None,
+        })
+        .filter_map(|impl_| impl_.trait_.as_ref());
+
+    // Limiting the creation of dummy traits to only those that are used by the lints.
+    // There are other foreign traits and it is not obvious how the dummy traits
+    // should look like for them.
+    let derivable_traits = [
+        "Debug",
+        "Clone",
+        "Copy",
+        "PartialOrd",
+        "Ord",
+        "PartialEq",
+        "Eq",
+        "Hash",
+        "Default",
+    ];
+    let auto_traits = ["Send", "Sync", "Unpin", "RefUnwindSafe", "UnwindSafe"];
+    let other_traits = ["Sized"];
+    let trait_names: Vec<&str> = vec![
+        derivable_traits.to_vec(),
+        auto_traits.to_vec(),
+        other_traits.to_vec(),
+    ]
+    .into_iter()
+    .flatten()
+    .collect();
+    let paths = paths.filter(|path| trait_names.contains(&path.name.as_str()));
+
+    paths
+        .map(|path| {
+            let dummy_item: Item = Item {
+                id: path.id.clone(),
+                crate_id: 0,
+                name: Some(path.name.clone()),
+                span: None,
+                visibility: rustdoc_types::Visibility::Public,
+                docs: None,
+                links: HashMap::new(),
+                attrs: Vec::new(),
+                deprecation: None,
+                inner: rustdoc_types::ItemEnum::Trait(rustdoc_types::Trait {
+                    is_auto: auto_traits.to_vec().contains(&path.name.as_str()),
```

Comment:
Let's not do the `.to_vec().contains()` bit, and instead do this properly: extract a function for making trait items, have it take an `is_auto: bool` parameter, etc.

To make it easier to define these traits, I'd consider making a new private (file-level) struct like `ManualTraitItem` with only the subset of fields we need / care about, and then pass instances of that struct to the method that makes the "real" rustdoc item from that "description".

Reply:
Done in 016398d8edbd2f7eb1dc87aebb0561ebb9ec957f.